src/schema/duration.rs

```diff
@@ -0,0 +1,358 @@
+use std::io::{Read, Write};
```

Comment:
Not sure if you need it or not, there is a nice crate called `chronos` that has pretty useful time/date functions.

Reply:
As it said in https://docs.rs/chrono/0.4.10/chrono/#duration
"Chrono currently uses the time::Duration type from the time crate to represent the magnitude of a time span"

And time::Duration does not follow ISO 8601. In particular, it stores weeks rather then months and years, and does not support Lexical representation.

I followed this, while implementing Duration type: https://www.w3.org/TR/xmlschema-2/#duration